### ISSUE TITLE
Wrong format for /proc/meminfo values

### ISSUE THREAD

**Issue by Edu4rdSHL** [Frogging-Family/linux-tkg#550]
**OS:** ArchLinux

**Bug description:** Somehow the kernel is putting leading zeros to most values in `/proc/meminfo` causing software such as `systemd-oomd` to fail. See https://github.com/systemd/systemd/issues/23912 for more details.

**Comment by Tk-Glitch** [comment]
I cannot reproduce the issue on my end.
```
Linux 5.18.9-262-tkg-pds #1 TKG SMP PREEMPT_DYNAMIC Mon, 04 Jul 2022 18:34:56 +0000 x86_64 GNU/Linux
```
```
 systemd-oomd[8833]: Memory used (33559547904) / total (33559547904) and swap used (13958987776) / total (15301836800) is more than 90.00%
 systemd-oomd[8833]: Failed to read memory.oom.group from /system.slice/systemd-udevd.service/udev, ignoring: No data available
 systemd-oomd[8833]: oomd attempting to kill 8853 with KILL
 systemd-oomd[8833]: oomd attempting to kill 8867 with KILL
 systemd-oomd[8833]: oomd attempting to kill 10209 with KILL
 systemd-oomd[8833]: oomd attempting to kill 8856 with KILL
 systemd-oomd[8833]: oomd attempting to kill 8859 with KILL
 systemd-oomd[8833]: oomd attempting to kill 8860 with KILL
 systemd-oomd[8833]: Considered 92 cgroups for killing, top candidates were:
 systemd-oomd[8833]:         Path: /user.slice/user-1000.slice/user@1000.service/app.slice/app-org.kde.konsole-90e9312619f04d95a80f7a090d5b36f2.scope
 systemd-oomd[8833]:                 Swap Usage: 12.0G
 systemd-oomd[8833]:         Path: /user.slice/user-1000.slice/user@1000.service/app.slice/app-firefox-dbd978c3aab04f4ba088a9b97c0b2af7.scope
 systemd-oomd[8833]:                 Swap Usage: 211.3M
 systemd-oomd[8833]:         Path: /user.slice/user-1000.slice/user@1000.service/session.slice/plasma-plasmashell.service
 systemd-oomd[8833]:                 Swap Usage: 126.5M
 systemd-oomd[8833]:         Path: /user.slice/user-1000.slice/user@1000.service/session.slice/plasma-kwin_x11.service
 systemd-oomd[8833]:                 Swap Usage: 92.9M
 systemd-oomd[8833]:         Path: /system.slice/packagekit.service
 systemd-oomd[8833]:                 Swap Usage: 68.0M
 systemd-oomd[8833]:         Path: /system.slice/sddm.service
 systemd-oomd[8833]:                 Swap Usage: 37.9M
 systemd-oomd[8833]:         Path: /user.slice/user-1000.slice/user@1000.service/app.slice/app-org.kde.ksysguard-eb3dd14a97384389b7092114153aaf42.scope
 systemd-oomd[8833]:                 Swap Usage: 36.8M
 systemd-oomd[8833]:         Path: /user.slice/user-1000.slice/user@1000.service/app.slice/app-org.kde.dolphin-b1822dff75434c38978e68220e12cfa5.scope
 systemd-oomd[8833]:                 Swap Usage: 36.7M
 systemd-oomd[8833]:         Path: /user.slice/user-1000.slice/user@1000.service/app.slice/app-org.kde.kate-587a6aaf533f4a2184777eaec833e857.scope
 systemd-oomd[8833]:                 Swap Usage: 36.3M
 systemd-oomd[8833]:         Path: /user.slice/user-1000.slice/user@1000.service/app.slice/app-org.kde.konsole-0e1f027c0fc04d559747435d852c20a1.scope
 systemd-oomd[8833]:                 Swap Usage: 28.0M
 systemd-oomd[8833]: Killed /user.slice/user-1000.slice/user@1000.service/app.slice/app-org.kde.konsole-90e9312619f04d95a80f7a090d5b36f2.scope due to memory used (33559547904) / total (33559547904) and swap used (13958987776) / total (15301836800) being more than 90.00%
 systemd[1252]: app-org.kde.konsole-90e9312619f04d95a80f7a090d5b36f2.scope: systemd-oomd killed 6 process(es) in this unit.
```

**Comment by Edu4rdSHL** [comment]
Can you try with the CFS scheduler?

**Comment by Tk-Glitch** [comment]
That was my first test actually. No leading zero in `/proc/meminfo` in both cases, and systemd-oomd works fine. Have you changed config? Anything fancy enabled?

**Comment by Tk-Glitch** [comment]
Built with Clang maybe?

**Comment by Edu4rdSHL** [comment]
I'm using/building the kernels for the [Chaotic AUR](https://aur.chaotic.cx/) repo, we use your PKGBUILD with [some changes](https://github.com/chaotic-aur/toolbox/blob/692146ee25731e2631a8bab8927f524586ffcb9e/src/lib/routines-tkg.sh#L55-L92) but nothing too aggressive (imo) and we only use Clang for building the `generic_v3_lto` variant.

I have tested the three kernel variants: `linux-tkg-cfs` (gcc), `linux-tkg-cfs-generic_v3` (gcc) and `linux-tkg-cfs-lto_v3` (clang) but the result is the same for all of them.

**Comment by LordDemecrius83** [comment]
I guess this is the same issue that causes this

![Screenshot from 2022-07-11 11-24-17](https://user-images.githubusercontent.com/89468157/178188801-006f43f3-af6d-485e-ae16-6bb6dde8261c.png)

**Comment by Tk-Glitch** [comment]
> I guess this is the same issue that causes this
> 
> This only happens on linux-tkg-*, GNOME System Monitor properly reports as 4GB on linux-zen
> 
> ![Screenshot from 2022-07-11 11-24-17](https://user-images.githubusercontent.com/89468157/178188801-006f43f3-af6d-485e-ae16-6bb6dde8261c.png)

Are you also using chaotic aur builds?

**Comment by LordDemecrius83** [comment]
> > I guess this is the same issue that causes this
> > This only happens on linux-tkg-*, GNOME System Monitor properly reports as 4GB on linux-zen
> > ![Screenshot from 2022-07-11 11-24-17](https://user-images.githubusercontent.com/89468157/178188801-006f43f3-af6d-485e-ae16-6bb6dde8261c.png)
> 
> Are you also using chaotic aur builds?

Yep

**Comment by Tk-Glitch** [comment]
I can indeed reproduce the issue with chaotic-aur builds. I cannot with my own builds. Similar configs used. Not sure what's going on here.

**Comment by LordDemecrius83** [comment]
> I can indeed reproduce the issue with chaotic-aur builds. I cannot with my own builds. Similar configs used. Not sure what's going on here.

I tried building it on my own which took 3.25 hours on my celeron laptop and GNOME System Monitor properly reports as 4GB and neofetch doesn't crash anymore

**Comment by Edu4rdSHL** [comment]
@pedrohlc is investigating the issue with the Chaotic Air builds as far as I know.

**Comment by qumaciel** [comment]
Any news on this issue?

**Comment by ptr1337** [comment]
Yes, I found it. Since the cachyos kernels are also precompiled available, a user reported this issue now also at us.

Since I saw that chaotic-aur does include also bcachefs into the compilation, I thought it can be the only issue why it affects also the cachyos kernels.
Compiled a kernel locally with bcachefs enabled and was able to reproduce this.

**Comment by PedroHLC** [comment]
TL;DR: BCachefs

**Comment by Tk-Glitch** [comment]
I have zero interest in it and never builds my kernels with it, that explains why I couldn't reproduce the issue. Thanks for reporting!

**Comment by iav** [comment]
fixed https://github.com/koverstreet/bcachefs/issues/447#issuecomment-1214449165
